### PR TITLE
Prefer `Array#sample` over `Kernel#rand` for move generation

### PR DESCRIPTION
**Pull request by splattael**
```ruby
require "benchmark/ips"

steps = (1..2).to_a

Benchmark.ips do |x|
  x.report "rand" do
    rand(2) + 1
  end

  x.report "sample" do
    steps.sample
  end
end
```

```
$ ruby foo.rb 
Warming up --------------------------------------
                rand   264.692k i/100ms
              sample   328.867k i/100ms
Calculating -------------------------------------
                rand      5.799M (± 7.9%) i/s -     28.851M in   5.018178s
              sample      9.404M (± 8.5%) i/s -     46.699M in   5.021198s
```

### CODE REVIEW

File: lib/mcts/examples/double_step.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ module Examples
     class DoubleStep
 
       FINAL_POSITION = 6
-      MAX_STEP = 2
+      STEPS = (1..2).to_a
```
Comment (PragTob) [inline lib/mcts/examples/double_step.rb]: I think a literal `[1, 2]` is easier to read here :)

Comment (splattael) [inline lib/mcts/examples/double_step.rb]: Sure, done.

**Comment by PragTob** [comment]
Thanks!

**Comment by splattael** [comment]
Again, thanks for merging :purple_heart:

**Pull request merged**